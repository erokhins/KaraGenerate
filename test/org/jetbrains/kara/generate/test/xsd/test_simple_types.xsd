<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:attributeGroup name="primitiveAttributes">
        <xsd:attribute name="boolAttr" type="xsd:boolean" default="true"/>
        <xsd:attribute name="stringAttr" type="xsd:string" default=""/>
        <xsd:attribute name="dateTimeAttr" type="xsd:dateTime" />
        <xsd:attribute name="stringWithoutType"/>
        <xsd:attribute name="floatAttr" type="xsd:float"/>
        <xsd:attribute name="integerAttr" type="xsd:integer"/>
        <xsd:attribute name="anyUriAttr" type="xsd:anyURI"/>
        <xsd:attribute name="positiveIntegerAttr" type="xsd:positiveInteger" default="1"/>
    </xsd:attributeGroup>

    <xsd:element name="simple">
        <xsd:complexType>
            <xsd:attributeGroup ref="primitiveAttributes"/>
            <xsd:attribute name="boolAttrInElement" type="xsd:boolean"/>
            <xsd:attribute name="strAttrInElement" type="xsd:string"/>
            <xsd:attribute name="noTypeStrAttrInElement" />
        </xsd:complexType>
    </xsd:element>

</xsd:schema>