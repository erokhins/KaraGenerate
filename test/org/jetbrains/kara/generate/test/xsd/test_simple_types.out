AttributeTypeDecl:
AttributeDecl:
   AttributeDecl = (anyUriAttr, null, null})
   AttributeTypeDecl = (anyUri, null)

   AttributeDecl = (boolAttr, null, true})
   AttributeTypeDecl = (boolean, null)

   AttributeDecl = (boolAttrInElement, null, null})
   AttributeTypeDecl = (boolean, null)

   AttributeDecl = (dateTimeAttr, null, null})
   AttributeTypeDecl = (dateTime, null)

   AttributeDecl = (floatAttr, null, null})
   AttributeTypeDecl = (float, null)

   AttributeDecl = (integerAttr, null, null})
   AttributeTypeDecl = (integer, null)

   AttributeDecl = (noTypeStrAttrInElement, null, null})
   AttributeTypeDecl = (string, null)

   AttributeDecl = (positiveIntegerAttr, null, 1})
   AttributeTypeDecl = (positiveInteger, null)

   AttributeDecl = (strAttrInElement, null, null})
   AttributeTypeDecl = (string, null)

   AttributeDecl = (stringAttr, null, })
   AttributeTypeDecl = (string, null)

   AttributeDecl = (stringWithoutType, null, null})
   AttributeTypeDecl = (string, null)

AttributeGroupDecl:
   AttributeGroup = primitiveAttributes
   ParentGroups = []
   NewAttributes =
      AttributeDecl = (anyUriAttr, null, null})
      AttributeTypeDecl = (anyUri, null)

      AttributeDecl = (boolAttr, null, true})
      AttributeTypeDecl = (boolean, null)

      AttributeDecl = (dateTimeAttr, null, null})
      AttributeTypeDecl = (dateTime, null)

      AttributeDecl = (floatAttr, null, null})
      AttributeTypeDecl = (float, null)

      AttributeDecl = (integerAttr, null, null})
      AttributeTypeDecl = (integer, null)

      AttributeDecl = (positiveIntegerAttr, null, 1})
      AttributeTypeDecl = (positiveInteger, null)

      AttributeDecl = (stringAttr, null, })
      AttributeTypeDecl = (string, null)

      AttributeDecl = (stringWithoutType, null, null})
      AttributeTypeDecl = (string, null)


ElementDecl:
   ElementDeclaration = (simple, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = [primitiveAttributes]
   NewAttributes =
      AttributeDecl = (boolAttrInElement, null, null})
      AttributeTypeDecl = (boolean, null)

      AttributeDecl = (noTypeStrAttrInElement, null, null})
      AttributeTypeDecl = (string, null)

      AttributeDecl = (strAttrInElement, null, null})
      AttributeTypeDecl = (string, null)


ElementGroup:
